### DISCUSSION TITLE
deep link open the trust wallet but not show the pop up to connect wallet

### DISCUSSION THREAD

**Discussion by arbaz-meo** [WalletConnect/walletconnect-monorepo#2367] [Developer Support] [answered]
i m using signclient   2.5.1 

creating the deeplink on node js server side and send this deep link to the mobile app through rest api . 

then its open the related wallet but not show any pop up for connect the wallet

**Comment by chendinh** [comment]
hmm i have no new idea, I got that error too. I just use web3modal and the deeplink work well again. https://docs.walletconnect.com/2.0/web3modal/about

**Comment by arbaz-meo** [comment]
@chendinh thanks lets wait for the wallet connect team to respond it

**Comment by arbaz-meo** [comment]
@0xAsimetriq can you help me ???

**Reply by xzilja** [reply]
Unable to replicate issue via several team members, which sdks / versions are you using for walletconnect / web3modal? Are you updated to latest TrustWallet version as well?

**Reply by arbaz-meo** [reply]
@0xAsimetriq i m using sign client only with version of 2.5.1 and all implementation done on node js level . so i m creating the wallet connect uri and sending to the mobile app from where i m redirecting to the trust wallet . so this procedure open the trust wallet but not shows the connect pop up. 

yes i update the trust wallet with highest version

**Reply by xzilja** [reply]
I think that could be the issue, sign-client is now on `2.7.3` I believe and introduced few upgrades to protocol, which trust wallet is likely already using.

**Reply by arbaz-meo** [reply]
while this way help me to connect the wallet through QR code @0xAsimetriq

**Reply by arbaz-meo** [reply]
@0xAsimetriq i try with updated version of signclient 2.7.3 but its not work for me

**Reply by xzilja** [reply]
In that case, could you provide your full implementation details i.e. code, nodejs details etc.

**Reply by arbaz-meo** [reply]
```
const  {SignClient} = require ('@walletconnect/sign-client');
const clientmap = new Map();
const uriMap = new Map();
const connectedMap = new Map();
const projectId = process.env.ProjectId;

const events = require('events');
const eventEmitter = new events.EventEmitter();




/*  
  we are initiliting the signclient for connecting wallet every time for every user on behalf of DeviceID 

*/

async function connectWallet(deviceId){
  try {
    let signClient = await SignClient.init({ projectId });
    const{uri,approval}=  await signClient.connect({
       requiredNamespaces: {
         eip155: {
           methods: ['eth_sign','eth_sendTransaction'],
           chains: ['eip155:80001'],
           events: ["connect",'accountsChanged']
         }
       },
       metadata: {
         name: 'Wallet',
         description: 'NFT Wallet',
         url: 'https://walletconnect.com/', // Do we have this one
         icons: ['https://avatars.githubusercontent.com/u/37784886'], // Looks like we need to deploy our splash somewhere
       
       },
    
      
     })
     console.log(uri);
     clientmap.set(deviceId,signClient);
     uriMap.set(deviceId,uri);
     connectedMap.set(deviceId,approval());

  } catch(e){
    console.log(e.message);
  }


}

module.exports.connectWallet=connectWallet;
module.exports.uriMap=uriMap;
module.exports.connectedMap=connectedMap;
module.exports.clientmap = clientmap;
```

@0xAsimetriq maybe this help you let me know if you need anything

**Reply by xzilja** [reply]
First thing that jumps out is running multiple sign clients in one environment, signClient was not designed for that, you'll likely need to switch to 1 instance of it and maintaining a map of deviceId's <-> pairings / sessions.

Any errors you are seeing as well?

**Reply by arbaz-meo** [reply]
@0xAsimetriq no i not see any error when  i stay with just wallet connect QR code

**Reply by arbaz-meo** [reply]
@0xAsimetriq  Asim   i can understand we need to have 1 instance but i need to connect multiple users  wallet with unreal engine . i try many sdk but its not working.  while this way is supporting our project and same time we can connect 100 of wallet so right now its cool till we not have good (authentic ) sdk for unreal engine . if you have any idea for implementation then you can suggest me

**Reply by xzilja** [reply]
You should still be able to do this, in fact even in a more efficient manner when using one instance of sign client:

1. You need to think of your server as a dapp connected to a big amount of wallets.
2. You will need to add logic on top of that where dapp creates new pairingTopic for unique user device id and store that somehow.
3. Then, when requests are made you can check pairing / session topics and map them back to device id I guess

**Reply by arbaz-meo** [reply]
@0xAsimetriq thanks i got it let me change the code and second  can you  help me this way will work to connect mobile wallet  with the help of deep link (walletct connect uri )

**Reply by arbaz-meo** [reply]
@0xAsimetriq i  move the code with  signclinet one instance 


```
const  {SignClient} = require ('@walletconnect/sign-client');
const uriMap = new Map();
const connectedMap = new Map();
const projectId = process.env.ProjectId;

let signClient;
async function intilizeSignClient() {
   signClient = await SignClient.init({ projectId });
} 



/*  
  we are paring session topic  every time for every user on behalf of DeviceID 

*/

async function connectWallet(deviceId){
  try {
   
    const{uri,approval}=  await signClient.connect({
       requiredNamespaces: {
         eip155: {
           methods: ['eth_sign','eth_sendTransaction'],
           chains: ['eip155:80001'],
           events: ["connect",'accountsChanged']
         }
       },
       metadata: {
         name: 'Wallet',
         description: 'NFT Wallet',
         url: 'https://walletconnect.com/', // Do we have this one
         icons: ['https://avatars.githubusercontent.com/u/37784886'], // Looks like we need to deploy our splash somewhere
       
       },
    
      
     })
     console.log(uri);
     uriMap.set(deviceId,uri);
     connectedMap.set(deviceId,approval());

  } catch(e){
    console.log(e.message);
  }


}

module.exports.connectWallet=connectWallet;
module.exports.uriMap=uriMap;
module.exports.connectedMap=connectedMap;
module.exports.signClient = signClient;
module.exports.intilizeSignClient = intilizeSignClient ; 
```

**Reply by arbaz-meo** [reply]
@0xAsimetriq can you help me to connect mobile wallet with deepl link

**Reply by xzilja** [reply]
Which bit exactly? Creation of connection uri? You can reference web3modal's code base for this, in particular this bit https://github.com/WalletConnect/web3modal/blob/V2/packages/core/src/utils/CoreUtil.ts#L40-L55

Here `appUrl` would be something like `metamask://`, `wcUri` is connection string you get from sdk i.e `wc:...` bit

**Reply by arbaz-meo** [reply]
@0xAsimetriq yes exactly creation of connection uri while i m getting wc uri  and i try to make like metamask:// "wcuri" and its open the wallet  but not open the pop for connect the wallet

**Reply by arbaz-meo** [reply, marked as answer]
@0xAsimetriq  you are amazing boy  to helping out .  so  trust wallet working for me i can connect wallet  by  wallet connect uri . 

wc:a9bfef6098dd0aa30d37aa776229290fc26765c63339ceb65126e0e0d372c03d@2?relay-protocol=irn&symKey=609c97aef498c1b77fb6be43513980f5ba25b6d549893a486e715be09493cd4a

**Reply by xzilja** [reply]
Nice 🎉 Glad to hear it worked out!

**Comment by glitch-txs** [comment]
Make sure to have the wallet updated as well*

**Reply by arbaz-meo** [reply]
@glitch-txs already updated
